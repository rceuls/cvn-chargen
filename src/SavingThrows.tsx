import { ICharacter } from './services/character.generator';

const SavingThrows = ({ character }: { character: ICharacter }) => {
  return (
    <>
      <div className="row">
        <div className="col-sm">
          <span className="font-weight-bold">Death Ray or Poison</span>
          <span className="ml-2">{character.charClass.savingThrows[0]}</span>
        </div>
      </div>
      <div className="row">
        <div className="col-sm">
          <span className="font-weight-bold">Magical Wands</span>
          <span className="ml-2">{character.charClass.savingThrows[1]}</span>
        </div>
      </div>
      <div className="row">
        <div className="col-sm">
          <span className="font-weight-bold">Paralysis or Petrification</span>
          <span className="ml-2">{character.charClass.savingThrows[2]}</span>
        </div>
      </div>
      <div className="row">
        <div className="col-sm">
          <span className="font-weight-bold">Dragon Breath</span>
          <span className="ml-2">{character.charClass.savingThrows[3]}</span>
        </div>
      </div>
      <div className="row">
        <div className="col-sm">
          <span className="font-weight-bold">Rods, Staves, Spells</span>
          <span className="ml-2">{character.charClass.savingThrows[4]}</span>
        </div>
      </div>

      <hr />
    </>
  );
};

export default SavingThrows;
