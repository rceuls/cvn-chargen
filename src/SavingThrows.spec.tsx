import renderer from 'react-test-renderer';
import SavingThrows from './SavingThrows';

it('renders correctly', () => {
  const tree = renderer
    .create(<SavingThrows savingThrows={[1, 2, 3, 4, 5]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
