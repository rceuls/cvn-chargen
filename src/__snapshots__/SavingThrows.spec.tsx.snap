// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Death Ray or Poison
      </span>
      <span
        className="ml-2"
      >
        1
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Magical Wands
      </span>
      <span
        className="ml-2"
      >
        2
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Paralysis or Petrification
      </span>
      <span
        className="ml-2"
      >
        3
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Dragon Breath
      </span>
      <span
        className="ml-2"
      >
        4
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Rods, Staves, Spells
      </span>
      <span
        className="ml-2"
      >
        5
      </span>
    </div>
  </div>,
  <hr />,
]
`;
